src/rest/APIRequest.js

```diff
@@ -46,11 +47,20 @@ class APIRequest {
       headers['Content-Type'] = 'application/json';
     }
 
+    const controller = new AbortController();
+    const timeout = setTimeout(() => controller.abort(), 10000);
```

Comment:
The timeout should probably be configurable (or even optional).
Maybe using `ClientOptions`? (Similar to [`retryLimit`])

[`retryLimit`]:https://github.com/discordjs/discord.js/blob/ddcc6cfec9ccb80d42e1d074bf03db2e0dae274e/src/util/Constants.js#L33-L53